plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -635,7 +417,7 @@ public ConnectorOutputTableHandle beginCreateTable(ConnectorSession session, Con
     public ConnectorInsertTableHandle beginInsert(ConnectorSession session, ConnectorTableHandle tableHandle)
     {
         IcebergTableHandle table = (IcebergTableHandle) tableHandle;
-        org.apache.iceberg.Table icebergTable = getIcebergTable(session, table.getSchemaTableName());
+        Table icebergTable = catalog.loadTable(session, table.getSchemaTableName());
```

Comment:
nit: we should be able to keep using `getIcebergTable` at all these places

Reply:
Is there an advantage to using it? If there is, then we should use it everywhere, otherwise we should remove the method entirely.

Reply:
yeah with the refactoring `getIcebergTable` is just calling the `catalog.loadTable`, that's why I directly removed the method. The only benefit I see for keeping the method is to minimize diff, which seems unnecessary to me.

Reply:
ah looks like we can't entirely remove the wrapper since the split manager needs it. this seems fine.